backend/prisma/schema.prisma

```diff
@@ -7,35 +7,238 @@ generator prisma_client {
+  PostTopics    PostTopics[]
+  Prompt        Prompt[]
+}
+
+model UserInterests {
+  id        Int      @default(autoincrement()) @id
+  user      User     @relation(fields: [userId], references: [id])
+  userId    Int
+  topic     Topic    @relation(fields: [topicId], references: [id])
+  topicId   Int
+  createdAt DateTime @default(now())
+
+  @@unique([userId, topicId])
+}
+
+model PostTopics {
+  id      Int   @default(autoincrement()) @id
+  post    Post  @relation(fields: [postId], references: [id])
+  postId  Int
+  topic   Topic @relation(fields: [topicId], references: [id])
+  topicId Int
+
+  @@unique([postId, topicId])
+}
+
+model PostLike {
+  id        Int      @default(autoincrement()) @id
+  createdAt DateTime @default(now())
+  author    User     @relation(fields: [userId], references: [id])
+  userId    Int
+  post      Post     @relation(fields: [postId], references: [id])
+  postId    Int
+}
+
+model CommentLike {
+  id        Int      @default(autoincrement()) @id
+  createdAt DateTime @default(now())
+  author    User     @relation(fields: [userId], references: [id])
+  userId    Int
+  comment   Comment  @relation(fields: [commentId], references: [id])
+  commentId Int
+}
+
+model Comment {
+  id        Int           @default(autoincrement()) @id
+  createdAt DateTime      @default(now())
+  updatedAt DateTime      @updatedAt
+  body      String
+  userId    Int
+  thread    Thread        @relation(fields: [threadId], references: [id])
+  threadId  Int
+  likes     CommentLike[]
+}
+
+model Thread {
+  id       Int       @default(autoincrement()) @id
+  position Int
+  range    Int
+  post     Post      @relation(fields: [postId], references: [id])
+  postId   Int
+  comments Comment[]
+}
+
+model Image {
+  id                 Int       @default(autoincrement()) @id
+  smallSize          String
+  largeSize          String
+  postId             String?
+  imageRole          ImageRole
+  Post               Post?     @relation(fields: [postId_ImageToPost], references: [id])
+  postId_ImageToPost Int?
+}
+
+model Prompt {
+  id      Int    @default(autoincrement()) @id
+  text    String
+  topic   Topic  @relation(fields: [topicId], references: [id])
+  topicId Int
+}
+
+model UserSubscription {
+  id        Int      @default(autoincrement()) @id
+  name      String
+  price     Int
+  user      User     @relation(fields: [userId], references: [id])
+  userId    Int
+  createdAt DateTime @default(now())
+  updatedAt DateTime @updatedAt
+}
+
+model PageView {
+  id                  Int      @default(autoincrement()) @id
+  user                User?    @relation(fields: [userId], references: [id])
+  userId              Int?
+  sessionId           String
+  pageName            String
+  timestamp           DateTime @default(now())
+  utmSource           String?
+  // TODO (robin-macpherson): find user agent string parsing library and store each part separately
+  user_agent_string   String
+  user_agent_language String
+}
```

Comment:
@nlitwin going to extract this `PageView` model out of here in a coming PR because all of our event data will be getting stored separately via a pipeline we'll be setting up specifically for that. But this shows the pattern of having a model per major event type for now 🙂